coverity/push_coverity_scan.py

```diff
@@ -0,0 +1,441 @@
+
+        if args.covtools is None:
+            # If no cov tools dir specified, then use a temporary (long path,
+            # given the need to redownload each time).
+            dir_path = mkdtemp()
+            tools_path = pathlib.Path(dir_path)
+            tools_path_set = True
+
+            download_coverity_scan_tools(logger, coverity_token, args.os, tools_path)
+        else:
+            # Coverity tools dir specified, see if it exists, contains tools and
+            # those tools are up to date.
+            tools_path = pathlib.Path(args.covtools)
+            tools_path = tools_path.resolve()
+            tools_path_set = True
+
+            if not tools_path.is_dir():
+
+                logger.log(logging.INFO, 'Tools dir does not exist, creating.')
+                tools_path.mkdir()
+
+                download_coverity_scan_tools(logger, coverity_token, args.os, tools_path)
+            else:
+                hash_path = tools_path / 'coverity_tool.md5'
+
+                if not hash_path.is_file():
+                    logger.log(logging.INFO, 'Hash file does not exist, re-downloading.')
+                    download_coverity_scan_tools(logger, coverity_token, args.os, tools_path)
+                else:
+                    # Attempt to check if our coverity scan package is up to date.
+                    if not check_coverity_scan_tools_version(coverity_token, args.os, tools_path):
+                        logger.log(logging.INFO, 'Hash file differs, re-downloading tools.')
+                        download_coverity_scan_tools(logger, coverity_token, args.os, tools_path)
+
+        with NamedTemporaryFile() as tar_file_handle:
+
+            tar_file = pathlib.Path(tar_file_handle.name)
+
+            if not mbedtls_path.is_dir():
+                raise ConfigError('MBed TLS directory specified does not exist.')
+
+            build_mbedtls(logger, mbedtls_path, tools_path, args.branch,
+                          args.pre_build_step, args.build_step, tar_file)
+
+            # Set this if you want to test without consuming the Coverity credit. The tar file will
+            # still obviously be built, and if you use a backup directory to save it, can still be
+            # manually uploaded via the website afterwards.
+            if not args.no_upload:
+                # send completed tar file to coverity
+                upload_build(logger, coverity_token, args.email, tar_file)
+
+            # If we want a backup of the tar file, then make one.
+            if args.backupdir is not None:
+                backup_path = pathlib.Path(args.backupdir)
+
+                if not backup_path.is_dir():
+                    logger.log(logging.INFO, 'Backup dir does not exist, creating.')
+                    backup_path.mkdir()
+
+                backup_path = backup_path.resolve()
+                backup_path = backup_path / datetime.today().strftime('mbedtls-%y-%m-%d.tar.gz')
```

Comment:
Non-blocker: a 2-digit year??
```suggestion
                backup_path = backup_path / datetime.today().strftime('mbedtls-%Y-%m-%d.tar.gz')
```